src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -112,11 +112,12 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
         }
         
         Map<Id,Account> primaryAccounts = new Map<Id,Account>([select id, npe01__One2OneContact__c, npe01__SYSTEM_AccountType__c from Account where id IN :oppAccounts]);
-        List<Contact> primaryContacts = [select AccountId,Account.Id,Account.npe01__SYSTEM_AccountType__c from Contact where Id IN :primaryContactIds];
-
-        for (Contact thisContact : primaryContacts)
-            if (thisContact.AccountId!=null)
-                contactsAndOneToOneAccounts.put(thisContact.Id,thisContact.Account);
+        if (!primaryContactIds.isEmpty()) {
+            for (Contact thisContact : [select AccountId,Account.Id,Account.npe01__SYSTEM_AccountType__c from Contact
+                    where Id IN :primaryContactIds AND AccountId != null]) {
```

Comment:
Good question. The change was to only issue the query when `primaryContactIds` is not empty - saving a query in cases when that collection is empty. This sort of relates to the quip doc we were commenting on this morning. Does filtering out null for Id fields help? QueryPlan seems to indicate that it's questionable, which probably doesn't match up with the note in the Quip doc. However, I think it would be safer to remove the "AND AccountId != null" clause for now and filter those out within the for loop.